From fd8daa6fd73abe560f7ff851e2c350b5025c482b Mon Sep 17 00:00:00 2001
From: Manjunath <manjunathx.b.s@intel.com>
Date: Mon, 5 Nov 2012 09:43:48 +0530
Subject: Music Playback Stops after encountering an erroneous stream in the playlist

BZ: 59379

This patch contains changes in MediaPlayerService.cpp. Currently,
for continuous playback, mNextClient->start is called only after
receiving MEDIA_PLAYBACK_COMPLETE msg. For erroneous streams,
AwesomePlayer notifies MEDIA_ERROR and this patch considers
that message as well in order to start mNextClient.

Category: AOSP Improvement
Domain: AUDIO_MEDIA-Audio Framework
Origin: Internal
Upstream-Candidate: Yes

Orig-Change-Id: I6201e519dae33d8e6d84e91ea3e740f4cb148eb7

Change-Id: I19ebf7c731ca910e1f0f01162ddc13025fd4eed7
Signed-off-by: Manjunath <manjunathx.b.s@intel.com>
Reviewed-on: http://android.intel.com:8080/82117
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 media/libmediaplayerservice/MediaPlayerService.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libmediaplayerservice/MediaPlayerService.cpp b/media/libmediaplayerservice/MediaPlayerService.cpp
index 0733d6c..464daa8 100644
--- a/media/libmediaplayerservice/MediaPlayerService.cpp
+++ b/media/libmediaplayerservice/MediaPlayerService.cpp
@@ -1092,7 +1092,7 @@ void MediaPlayerService::Client::notify(
     {
         Mutex::Autolock l(client->mLock);
         c = client->mClient;
-        if (msg == MEDIA_PLAYBACK_COMPLETE && client->mNextClient != NULL) {
+        if ((msg == MEDIA_PLAYBACK_COMPLETE || msg == MEDIA_ERROR) && client->mNextClient != NULL) {
             if (client->mAudioOutput != NULL)
                 client->mAudioOutput->switchToNextOutput();
             client->mNextClient->start();
-- 
1.7.4.1

